scripts/publish_github_stats.js

```diff
@@ -0,0 +1,145 @@
+
+    if (issue.pull_request) {
+      openPullRequestsTotal++;
+      if (!isMapboxAuthor) {
+        openPullRequestsTotalFromNonMembers++;
+      }
+      if (issueCreatedAt >= oneMonthAgo) {
+        openPullRequestsSinceLastMonth++;
+        if (!isMapboxAuthor) {
+          openPullRequestsSinceLastMonthFromNonMembers++;
+        }
+      }
+    } else {
+      openIssuesTotal++;
+      if (!isMapboxAuthor) {
+        openIssuesTotalFromNonMembers++;
+      }
+      issue.labels.forEach(function (label) {
+        switch (label.name) {
+          case "Core":
+            openIssuesTotalCore++;
+            break;
+          case "Android":
+            openIssuesTotalAndroid++;
+            break;
+          case "iOS":
+            openIssuesTotalIOS++;
+            break;
+          case "GL JS parity":
+            openIssuesTotalGLJSParity++;
+            break;
+          default:
+            break;
+        }
+      });
+    }
+  });
+}
+
+function publishMetrics() {
+  let metrics = {
+    'created_at': new Date().toISOString().substring(0, 10),
+    'open_issues_total': openIssuesTotal,
+    'open_issues_total_from_non_members': openIssuesTotalFromNonMembers,
+    'open_issues_total_core': openIssuesTotalCore,
+    'open_issues_total_android': openIssuesTotalAndroid,
+    'open_issues_total_ios': openIssuesTotalIOS,
+    'open_issues_total_gl_js_parity': openIssuesTotalGLJSParity,
+    'open_pull_requests_total': openPullRequestsTotal,
+    'open_pull_requests_total_from_non_members': openPullRequestsTotalFromNonMembers,
+    'open_pull_requests_since_last_month': openPullRequestsSinceLastMonth,
+    'open_pull_requests_since_last_month_from_non_members': openPullRequestsSinceLastMonthFromNonMembers
+  };
+
+  var promise = new AWS.S3({region: 'us-east-1'}).putObject({
+    Body: zlib.gzipSync(JSON.stringify(metrics)),
+    Bucket: 'mapbox-loading-dock',
+    Key: `raw/mobile_staging.github_stats/${metrics['created_at']}/METRIC.json.gz`,
+    CacheControl: 'max-age=300',
+    ContentEncoding: 'gzip',
+    ContentType: 'application/json'
+  }).promise();
+
+  return Promise.all([promise]).then(data => {
+    return console.log("Successfully uploaded Github Stats metrics to S3");
+  }).catch(err => {
+    console.log("Error uploading Github Stats metrics to S3 " + err.message);
+    return err;
+  });
+}
+
+function recursiveGetForRepo(query) {
+  assert(query);
+  query.then(result => {
+    collectMetricsFromIssues(result);
+    if (github.hasNextPage(result)) {
+      recursiveGetForRepo(github.getNextPage(result));
+    } else {
+      publishMetrics();
+    }
+  });
+}
+
+github.apps.createInstallationToken({ installation_id: SIZE_CHECK_APP_INSTALLATION_ID })
+  .then(({data}) => {
+    github.authenticate({ type: 'token', token: data.token });
+  })
+  .then(() => {
+    recursiveGetForRepo(github.issues.listForRepo({ owner: 'mapbox', repo: 'mapbox-gl-native', state: 'open', per_page: 100 }));
```

Comment:
`github.issues.listForRepo` returns promise, maybe worth adding catch block in `recursiveGetForRepo`